### PR TITLE
Refact invite handler

### PR DESCRIPTION
**Pull request by raonismaneoto**
**Feature/Bug description:**
The invite_handler was handling with invite_user and invite_institution. Low cohesion problem.
**Solution:**
I've created invite_user_handler and invite_institution_handler and split the invite_handler's logic to them.

**TODO/FIXME:** n/a

### CODE REVIEW

File: backend/handlers/invite_handler.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-"""Invite Handler."""
+"""Invite User Handler."""
```
Comment (Luiz-FS) [inline backend/handlers/invite_handler.py]: Invite Handler

File: backend/handlers/__init__.py [hunk 2]
```diff
@@ -56,7 +58,8 @@
     search_handler, subscribe_post_handler, user_handler,
     user_institutions_handler, user_profile_handler,
     user_request_collection_handler, user_timeline_handler, vote_handler,
-    invite_hierarchy_collection_handler, invite_user_collection_handler
+    invite_hierarchy_collection_handler, invite_user_collection_handler,
+    invite_institution_handler, invite_handler
```
Comment (mayzabeel) [inline backend/handlers/__init__.py]: insert invite_user_handler

File: backend/handlers/invite_handler.py (hunk 1 above)
Comment (mayzabeel) [inline backend/handlers/invite_handler.py]: REmove User

File: backend/test/invite_handler_test.py [hunk 3]
```diff
@@ -1,22 +1,21 @@
 # -*- coding: utf-8 -*-
-"""Invite Handler Test."""
+"""Invite User Handler Test."""
```
Comment (Luiz-FS) [inline backend/test/invite_handler_test.py]: Invite Handler Test

File: backend/test/invite_handler_test.py [hunk 4]
```diff
@@ -1,22 +1,21 @@
...
 
 import json
 import mocks
 
 from test_base_handler import TestBaseHandler
-from search_module import SearchInstitution
 from models import InviteUser
 from models import Invite
-from models import InviteInstitution
 from handlers.invite_handler import InviteHandler
```
Comment (Luiz-FS) [inline backend/test/invite_handler_test.py]: from handlers import InviteHandler

File: backend/test/invite_handler_test.py [hunk 5]
```diff
@@ -1,22 +1,21 @@
...
 from mock import patch
 
 CURRENT_INSTITUTION = {'name': 'currentInstitution'}
 CURRENT_INST_STRING = json.dumps(CURRENT_INSTITUTION)
 
+
 class InviteHandlerTest(TestBaseHandler):
-    """Invite Handler Test."""
+    """Invite User Handler Test."""
```
Comment (Luiz-FS) [inline backend/test/invite_handler_test.py]: same

File: backend/test/invite_handler_test.py [hunk 6]
```diff
@@ -54,10 +53,9 @@ def setUp(cls):
         cls.invite = InviteUser.create(cls.data)
         cls.invite.put()
 
-
     @patch('util.login_service.verify_token', return_value={'email': 'otheruser@test.com'})
     def test_get(self, verify_token):
-        """Test method get of InviteHandler."""
+        """Test method get of InviteUserHandler."""
```
Comment (Luiz-FS) [inline backend/test/invite_handler_test.py]: same

File: backend/test/invite_institution_handler_test.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+"""Invite User Handler Test."""
```
Comment (Luiz-FS) [inline backend/test/invite_institution_handler_test.py]: same

File: backend/test/invite_institution_handler_test.py [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+from models import Invite
+from handlers import InviteInstitutionHandler
+from mock import patch
+
+CURRENT_INSTITUTION = {'name': 'currentInstitution'}
+CURRENT_INST_STRING = json.dumps(CURRENT_INSTITUTION)
+
+
+class InviteInstitutionHandlerTest(TestBaseHandler):
+    """Invite User Handler Test."""
```
Comment (Luiz-FS) [inline backend/test/invite_institution_handler_test.py]: same

File: backend/test/invite_institution_handler_test.py [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+            'institution_key': cls.inst_test.key.urlsafe(),
+            'type_of_invite': 'USER',
+            'suggestion_institution_name': 'test'
+        }
+        cls.invite = InviteInstitution.create(cls.data)
+        cls.invite.put()
+
+    @patch.object(Invite, 'send_notification')
+    @patch('util.login_service.verify_token', return_value={'email': 'otheruser@test.com'})
+    def test_delete(self, verify_token, send_notification):
```
Comment (Luiz-FS) [inline backend/test/invite_institution_handler_test.py]: add docstring

File: backend/test/invite_handler_test.py (hunk 3 above)
Comment (mayzabeel) [inline backend/test/invite_handler_test.py]: remove user

File: backend/test/invite_handler_test.py (hunk 5 above)
Comment (mayzabeel) [inline backend/test/invite_handler_test.py]: remove user

File: backend/test/invite_handler_test.py (hunk 6 above)
Comment (mayzabeel) [inline backend/test/invite_handler_test.py]: InviteHandler

File: backend/handlers/__init__.py [hunk 10]
```diff
@@ -39,6 +39,8 @@
 from .vote_handler import *
 from .invite_hierarchy_collection_handler import *
 from .invite_user_collection_handler import *
+from .invite_institution_handler import *
+from .invite_user_handler import *
```
Comment (Luiz-FS) [inline backend/handlers/__init__.py]: Add in handlers list

File: backend/test/invite_institution_handler_test.py (hunk 7 above)
Comment (mayzabeel) [inline backend/test/invite_institution_handler_test.py]: Invite Institution

File: backend/test/invite_institution_handler_test.py (hunk 8 above)
Comment (mayzabeel) [inline backend/test/invite_institution_handler_test.py]: same

File: backend/test/invite_institution_handler_test.py [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+        """Test method delete of InviteUserHandler."""
```
Comment (mayzabeel) [inline backend/test/invite_institution_handler_test.py]: Institution

File: backend/handlers/invite_institution_handler.py [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+    """Invite Institution Handler."""
+
+    @json_response
+    @login_required
+    def delete(self, user, invite_urlsafe):
+        """Change invite status from 'sent' to 'rejected'.
+        
+        This method is called when an user reject a invite
+        to create a new institution.
+        """
```
Comment (Luiz-FS) [inline backend/handlers/invite_institution_handler.py]: Make sure you receive the correct invitation

File: backend/handlers/invite_user_handler.py [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+class InviteUserHandler(BaseHandler):
+    """Invite User Handler."""
+
+    @json_response
+    @login_required
+    def delete(self, user, invite_urlsafe):
+        """Change invite status from 'sent' to 'rejected'.
+        This method is called when a user reject an invite
+        to be member of an institution.        
+        """
```
Comment (Luiz-FS) [inline backend/handlers/invite_user_handler.py]: Make sure you receive the correct invitation

File: backend/handlers/invite_user_handler.py [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+        invite.send_response_notification(user.current_institution, user.key, 'REJECT')
+
+    @json_response
+    @login_required
+    @ndb.transactional(xg=True)
+    def patch(self, user, invite_urlsafe):
+        """Handle PATCH Requests.
+        This method is called when an user accept
+        the invite to be a member of an institution.
+        """
```
Comment (Luiz-FS) [inline backend/handlers/invite_user_handler.py]: same

File: backend/test/invite_institution_handler_test.py [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+    INVITE_INSTITUTION_URI = "/api/invites/institution/(.*)"
+
+    @classmethod
+    def setUp(cls):
+        """Provide the base for the tests."""
+        super(InviteInstitutionHandlerTest, cls).setUp()
+
+        methods = set(cls.webapp2.WSGIApplication.allowed_methods)
+        methods.add('PATCH')
+        cls.webapp2.WSGIApplication.allowed_methods = frozenset(methods)
```
Comment (Luiz-FS) [inline backend/test/invite_institution_handler_test.py]: Is this necessary for this test?

File: backend/handlers/invite_institution_handler.py [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+        invite_key = ndb.Key(urlsafe=invite_urlsafe)
+        invite = invite_key.get()
+
+        invite_class_name = invite.__class__.__name__
+        Utils._assert(invite_class_name != 'InviteInstitution',
+                      "The invite's type is not the expected one",
```
Comment (mayzabeel) [inline backend/handlers/invite_institution_handler.py]: This message doesn't help us to solve. Be more specific ...The invite's type expected is ... but receive ...

**Pull request merged**